snp_application/src/bt/snp_bt_ros_nodes.cpp

```diff
@@ -154,16 +154,25 @@ BT::NodeStatus StartReconstructionServiceNode::onResponseReceived(const typename
 
 bool StopReconstructionServiceNode::setRequest(typename Request::SharedPtr& request)
 {
-  request->archive_directory = "";
+  request->archive_directory = get_parameter_or<std::string>(node_, IR_ARCHIVE_DIR_PARAM, "");
   request->mesh_filepath = get_parameter<std::string>(node_, MESH_FILE_PARAM);
-  request->min_num_faces = 1000;
-
-  industrial_reconstruction_msgs::msg::NormalFilterParams norm_filt;
-  norm_filt.normal_direction.x = 0;
-  norm_filt.normal_direction.y = 0;
-  norm_filt.normal_direction.z = 1;
-  norm_filt.angle = 85;
-  request->normal_filters.push_back(norm_filt);
+  request->min_num_faces = get_parameter<int>(node_, IR_MIN_FACES_PARAM);
+
+  try
+  {
+    industrial_reconstruction_msgs::msg::NormalFilterParams norm_filt;
+    norm_filt.angle = get_parameter<double>(node_, IR_NORMAL_ANGLE_TOL_PARAM);
+    norm_filt.normal_direction.x = get_parameter<double>(node_, IR_NORMAL_X_PARAM);
+    norm_filt.normal_direction.y = get_parameter<double>(node_, IR_NORMAL_Y_PARAM);
+    norm_filt.normal_direction.z = get_parameter<double>(node_, IR_NORMAL_Z_PARAM);
+
+    // Do not add a normal filter if the angle is less than 0.0
+    if (norm_filt.angle > 0.0)
+      request->normal_filters.push_back(norm_filt);
+  }
+  catch(const std::exception&)
+  {
+  }
```

Comment:
Why is this in a try-catch?

Reply:
In the case that the parameter can't be found, I want to continue without it. I'm not sure this would ever happen since we declare all of them with a default value, but may as well keep it

Reply:
I guess if someone overrode the constructor and didn't declare them that could happen.